### PR TITLE
Longhaul Tests: Add filter to log upload to avoid uploading entire logs periodically

### PR DESCRIPTION
**Pull request by and-rewsmith**
Currently we upload the entire logs every 10 minutes which leads to resource issues on our test devices. This PR limits the log upload duration to the upload frequency plus some buffer to avoid uploading logs un-necessarily. This will fix the resource issues on longhaul agents.

### CODE REVIEW

File: test/modules/TestResultCoordinator/Services/TestResultReportingService.cs [hunk 1]
```diff
@@ -103,7 +103,10 @@ async void DoWorkAsync(object state)
                 {
                     Uri blobContainerWriteUriForLog = await TestReportUtil.GetOrCreateBlobContainerSasUriForLogAsync(this.serviceSpecificSettings.StorageAccountConnectionString);
                     blobContainerUri = $"{blobContainerWriteUriForLog.Scheme}{Uri.SchemeDelimiter}{blobContainerWriteUriForLog.Authority}{blobContainerWriteUriForLog.AbsolutePath}";
-                    await TestReportUtil.UploadLogsAsync(Settings.Current.IoTHubConnectionString, blobContainerWriteUriForLog, this.logger);
+
+                    // Add buffer to duration to ensure we capture all logs
+                    TimeSpan uploadLogDuration = this.sendReportFrequency + TimeSpan.FromMinutes(10);
```
Comment (damonbarry) [inline test/modules/TestResultCoordinator/Services/TestResultReportingService.cs:108]: It looks like `this.sendReportFrequency` is set to -1ms in the connectivity case. Will this math (-1ms + 10min) cause problems?

Comment (and-rewsmith) [reply test/modules/TestResultCoordinator/Services/TestResultReportingService.cs:108]: Ah yes this will cause issues. Thinking about the best fix now.

File: test/modules/TestResultCoordinator/Services/TestResultReportingService.cs [hunk 2]
```diff
@@ -39,16 +40,19 @@ public TestResultReportingService(ITestOperationResultStorage storage)
...
                         // Set sendReportFrequency to -1ms to indicate that the sending report Timer won't repeat
                         this.sendReportFrequency = TimeSpan.FromMilliseconds(-1);
+                        this.logUploadDuration = Option.None<TimeSpan>();
                         break;
                     }
 
                 case TestMode.LongHaul:
                     {
+                        // In long haul mode, wait 1 report frequency before starting
+                        this.delayBeforeWork = this.sendReportFrequency;
```
Comment (damonbarry) [inline test/modules/TestResultCoordinator/Services/TestResultReportingService.cs:50]: Why did you move this? You're reading the value of `this.sendReportFrequency` _before_ you set it (the next line).

Comment (and-rewsmith) [reply test/modules/TestResultCoordinator/Services/TestResultReportingService.cs:50]: Wanted to make the assignment order consistent with above block. Got confused as `Settings.Current` has its own `SendReportFrequency`. Changed it back.

Will request review when both connectivity and longhaul are passing.

File: test/modules/TestResultCoordinator/TestReportUtil.cs [hunk 3]
```diff
@@ -154,9 +154,21 @@ internal static async Task UploadLogsAsync(string iotHubConnectionString, Uri bl
...
+
             CloudToDeviceMethod uploadLogRequest =
-                new CloudToDeviceMethod("UploadModuleLogs")
-                .SetPayloadJson($"{{ \"schemaVersion\": \"1.0\", \"sasUrl\": \"{blobContainerWriteUri.AbsoluteUri}\", \"items\": [{{ \"id\": \".*\", \"filter\": {{}} }}], \"encoding\": \"gzip\", \"contentType\": \"json\" }}");
+                new CloudToDeviceMethod("UploadModuleLogs");
+            uploadLogRequest = logUploadDuration.Match<CloudToDeviceMethod>(
+            (TimeSpan logUploadDuration) =>
+            {
+                DateTime logStart = DateTime.UtcNow - logUploadDuration;
+                uploadLogRequest = uploadLogRequest.SetPayloadJson($"{{ \"schemaVersion\": \"1.0\", \"sasUrl\": \"{blobContainerWriteUri.AbsoluteUri}\", \"items\": [{{ \"id\": \".*\", \"filter\": {{\"since\": \"{logStart.ToString("O")}\"}} }}], \"encoding\": \"gzip\", \"contentType\": \"json\" }}");
```
Comment (darobs) [inline test/modules/TestResultCoordinator/TestReportUtil.cs:164]: You could just return from line 164, but this is a pretty minor stylistic change.

Comment (darobs) [reply test/modules/TestResultCoordinator/TestReportUtil.cs:164]: Same for line 169.

**Review by darobs — DISMISSED** [review]
code style comments only.

**Review by darobs — APPROVED** [review]
🏃
